embedded-graphics/MIGRATING-0.5-0.6.md

```diff
@@ -0,0 +1,444 @@
+- `draw_pixel()` now returns `Result<(), Self::Error>` to account for driver error handling.
+
+New methods have been added to `DrawTarget` to enable accelerated drawing operations if the hardware supports it:
+
+- `DrawTarget::draw_circle` - draw a circle using hardware accelerated operations.
+- `DrawTarget::draw_rectangle` - draw a rectangle using hardware accelerated operations.
+- `DrawTarget::draw_triangle` - draw a triangle using hardware accelerated operations.
+- `DrawTarget::draw_line` - draw a line using hardware accelerated operations.
+- `DrawTarget::draw_image` - draw an image using hardware accelerated operations.
+
+By default these delegate to `DrawTarget::draw_iter` but can be overridden with custom implementations for improved draw performance.
+
+More information is provided in the [`DrawTarget` trait documentation](https://docs.rs/embedded-graphics/0.6.0/embedded_graphics/trait.DrawTarget.html).
+
+### Choosing the right pixel color type
+
+Below are some common use cases to help choose the right pixel color for a given display module. The full list of pixel colors is described in [the table above](#pixel-colors).
+
+#### `BinaryColor`
+
+If the display only supports two states, use `BinaryColor`. This is applicable to monochrome OLED displays like the SSD1306 or SH1106, character/bitmap LCDs and even LED matrices.
+
+#### `Rgb565`
+
+Use `Rgb565` for displays advertised as 16 bit color displays will often use a 5R 6G 5B pixel packing, storing each pixel color as a `u16`.
+
+#### `Bgr565`
+
+If the display's pixel order is reversed (BGR instead of RGB) and cannot be changed in configuration, use this type instead of `Rgb565`.
+
+#### `Rgb888`
+
+24 bit color, most useful in the embedded graphics simulator, or for high color display modules.
+
+## For font authors
+
+The `FontBuilderConf` trait is renamed to `Font`. This is the only trait impl that is required for embedded-graphics integration. `FontBuilder` is now removed, along with the `embedded_graphics::fonts::font_builder` module.
+
+The `Font` trait implementation has changed slightly, replacing `CHAR_WIDTH` and `CHAR_HEIGHT` with a single `CHARACTER_SIZE` constant, using the `Size` struct:
+
+```diff
+- use embedded_graphics::fonts::font_builder::{FontBuilder, FontBuilderConf};
++ use embedded_graphics::fonts::Font;
+
++ use embedded_graphics::geometry::Size;
+
+  #[derive(Debug, Copy, Clone)]
+  pub struct MyFont;
+
+  impl Font for MyFont {
+      const FONT_IMAGE: &'static [u8] = include_bytes!("../data/my_font.raw");
+-     const CHAR_WIDTH: u32 = 12;
+-     const CHAR_HEIGHT: u32 = 22;
++     const CHARACTER_SIZE: Size = Size::new(12, 22);
+      const FONT_IMAGE_WIDTH: u32 = 480;
+  }
+```
+
+The `CHARACTER_SPACING` and `VARIABLE_WIDTH` constants were also added, creating support for basic variable width fonts.
```

Comment:
`CHARACTER_SPACING` can also be used for fixed fonts, like in the `custom-font' example.